sdl-core/src/main/scala/io/smartdatalake/util/misc/SmartDataLakeLogger.scala

```diff
@@ -38,9 +39,11 @@ trait SmartDataLakeLogger {
     e
   }
 
-  private[smartdatalake] def logWithSeverity(severity: Level, msg: String): Unit = {
+  private[smartdatalake] def logWithSeverity(severity: Level, msg: String, cause: Throwable): Unit = {
     severity match {
+      case Level.ERROR if Environment.includeDAGResultExceptionInLog => logger.error(msg, cause)
       case Level.ERROR => logger.error(msg)
```

Comment:
Isn't log.error called twice if includeDAGResultExceptionInLog is true, once with cause, once without?

Reply:
No, pattern matching always executes only one case statement and then stops.